state/unit.go

```diff
@@ -1454,12 +1458,32 @@ func (u *Unit) SetStatus(unitStatus status.StatusInfo) error {
 }
 
 // OpenedPortRanges returns a UnitPortRanges object that can be used to query
+// and/or mutate the port ranges opened by the unit.
+func (u *Unit) OpenedPortRanges() (UnitPortRanges, error) {
+	if u.ShouldBeAssigned() {
+		return u.openedPortRangesForMachine()
```

Comment:
"For Machine" is confusing here because even though the ports are stored against the machine, we are just fetching what ports this iaas unit has opened, machine really has nothing to do with the result we are returning.

Reply:
This is existing code, we do need to return ports opened on the unit's machine for the uniter to do ports conflict check.